src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/ReactiveCosmosTemplate.java

```diff
@@ -0,0 +1,497 @@
+     * @param entityClass   the entity class
+     * @param containerName the container name
+     * @return Mono with a boolean or error
+     */
+    @Override
+    public Mono<Boolean> exists(DocumentQuery query, Class<?> entityClass, String containerName) {
+        return getCountValue(query, true, containerName).flatMap(count -> Mono.just(count > 0));
+    }
+
+    /**
+     * Exists
+     * @param id the id
+     * @param entityClass the entity class
+     * @param containerName the containercontainer nam,e
+     * @return
+     */
+    public Mono<Boolean> existsById(Object id, Class<?> entityClass, String containerName) {
+        return findById(containerName, id, entityClass)
+                .flatMap(o -> Mono.just(o !=  null));
+    }
+
+    /**
+     * Count
+     *
+     * @param containerName the container name
+     * @return Mono with the count or error
+     */
+    @Override
+    public Mono<Long> count(String containerName) {
+        final DocumentQuery query = new DocumentQuery(Criteria.getInstance(CriteriaType.ALL));
+        return getCountValue(query, true, containerName);
+    }
+
+    /**
+     * Count
+     *
+     * @param query         the document query
+     * @param containerName the container name
+     * @return Mono with count or error
+     */
+    @Override
+    public Mono<Long> count(DocumentQuery query, String containerName) {
+        return getCountValue(query, true, containerName);
+    }
+
+    private Mono<Long> getCountValue(DocumentQuery query, boolean isCrossPartitionQuery, String containerName) {
+        final SqlQuerySpec querySpec = new CountQueryGenerator().generateCosmos(query);
+        final FeedOptions options = new FeedOptions();
+
+        options.setEnableCrossPartitionQuery(isCrossPartitionQuery);
+
+        return executeQuery(querySpec, containerName, options)
+                .onErrorResume(this::databaseAccessExceptionHandler)
+                .next()
+                .map(r -> r.getResults().get(0).getLong(COUNT_VALUE_KEY));
+    }
+
+    private Flux<FeedResponse<CosmosItem>> executeQuery(SqlQuerySpec sqlQuerySpec, String collectionName,
+                                                        FeedOptions options) {
+
+        return getCosmosClient().getDatabase(this.databaseName)
+                .getContainer(collectionName)
+                .queryItems(sqlQuerySpec, options);
+    }
+
+    private <T> Mono<T> databaseAccessExceptionHandler(Throwable e) {
+        throw new DocumentDBAccessException("failed to access cosmosdb database", e);
+    }
+
+    /**
+     * Delete container with container name
+     *
+     * @param containerName the container name
+     */
+    @Override
+    public void deleteContainer(@NonNull String containerName) {
+        Assert.hasText(containerName, "containerName should have text.");
+        try {
+            getCosmosClient().getDatabase(this.databaseName).getContainer(containerName).delete().block();
```

Comment:
Never use blocking operations in an API that exposes reactive calls. Calling this method and blocking underlying threads is almost a guarantee to defunct the entire application.